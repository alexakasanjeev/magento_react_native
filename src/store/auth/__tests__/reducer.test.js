import authReducer, { INITIAL_STATE } from '../reducer';
import {
  MAGENTO,
  ACTION_USER_LOGOUT,
  RESET_AUTH_STATE
} from '../../../constants';

describe('auth reducer', () => {
  test('should return the initial state', () => {
    // Exercise
    const newState = authReducer(undefined, {});

    // verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle sign in loading', () => {
    // Setup
    const action = {
      type: MAGENTO.SIGN_IN_LOADING,
    };

    // Exercise
    const newState = authReducer(INITIAL_STATE, action);

    // verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle sign in success', () => {
    // Setup
    const action = {
      type: MAGENTO.SIGN_IN_SUCCESS,
    };

    // Exercise
    const newState = authReducer(INITIAL_STATE, action);

    // verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle sign in failure', () => {
    // Setup
    const action = {
      type: MAGENTO.SIGN_IN_FAILURE,
      payload: {
        errorMessage: 'Unable to sign in, wrong credentials'
      }
    };

    // Exercise
    const newState = authReducer(INITIAL_STATE, action);

    // verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle sign up loading', () => {
    // Setup
    const action = {
      type: MAGENTO.SIGN_UP_LOADING,
    };

    // Exercise
    const newState = authReducer(INITIAL_STATE, action);

    // verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle sign up success', () => {
    // Setup
    const action = {
      type: MAGENTO.SIGN_UP_SUCCESS,
    };

    // Exercise
    const newState = authReducer(INITIAL_STATE, action);

    // verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle sign up failure', () => {
    // Setup
    const action = {
      type: MAGENTO.SIGN_UP_FAILURE,
      payload: {
        errorMessage: 'Unable to sign up, invalid credentials'
      }
    };

    // Exercise
    const newState = authReducer(INITIAL_STATE, action);

    // verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle reset password loading', () => {
    // Setup
    const action = {
      type: MAGENTO.RESET_PASSWORD_LOADING,
    };

    // Exercise
    const newState = authReducer(INITIAL_STATE, action);

    // verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle reset password success', () => {
    // Setup
    const action = {
      type: MAGENTO.RESET_PASSWORD_SUCCESS,
    };

    // Exercise
    const newState = authReducer(INITIAL_STATE, action);

    // verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle reset password failure', () => {
    // Setup
    const action = {
      type: MAGENTO.RESET_PASSWORD_FAILURE,
      payload: {
        errorMessage: 'Unable to reset password, invalid email'
      }
    };

    // Exercise
    const newState = authReducer(INITIAL_STATE, action);

    // verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle action user logout', () => {
    // Setup
    const action = {
      type: ACTION_USER_LOGOUT,
    };

    // Exercise
    const newState = authReducer(INITIAL_STATE, action);

    // verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle reset auth state', () => {
    // Setup
    const action = {
      type: RESET_AUTH_STATE,
    };

    // Exercise
    const newState = authReducer(INITIAL_STATE, action);

    // verify
    expect(newState).toMatchSnapshot();
  });
});
