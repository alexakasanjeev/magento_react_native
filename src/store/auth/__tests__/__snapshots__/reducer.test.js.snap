// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auth reducer should handle action user logout 1`] = `
Object {
  "resetPasswordErrorMessage": "",
  "resetPasswordStatus": "NOT_STARTED",
  "signInErrorMessage": "",
  "signInStatus": "NOT_STARTED",
  "signUpErrorMessage": "",
  "signUpStatus": "NOT_STARTED",
}
`;

exports[`auth reducer should handle reset auth state 1`] = `
Object {
  "resetPasswordErrorMessage": "",
  "resetPasswordStatus": "NOT_STARTED",
  "signInErrorMessage": "",
  "signInStatus": "NOT_STARTED",
  "signUpErrorMessage": "",
  "signUpStatus": "NOT_STARTED",
}
`;

exports[`auth reducer should handle reset password failure 1`] = `
Object {
  "resetPasswordErrorMessage": "Unable to reset password, invalid email",
  "resetPasswordStatus": "ERROR",
  "signInErrorMessage": "",
  "signInStatus": "NOT_STARTED",
  "signUpErrorMessage": "",
  "signUpStatus": "NOT_STARTED",
}
`;

exports[`auth reducer should handle reset password loading 1`] = `
Object {
  "resetPasswordErrorMessage": "",
  "resetPasswordStatus": "LOADING",
  "signInErrorMessage": "",
  "signInStatus": "NOT_STARTED",
  "signUpErrorMessage": "",
  "signUpStatus": "NOT_STARTED",
}
`;

exports[`auth reducer should handle reset password success 1`] = `
Object {
  "resetPasswordErrorMessage": "",
  "resetPasswordStatus": "SUCCESS",
  "signInErrorMessage": "",
  "signInStatus": "NOT_STARTED",
  "signUpErrorMessage": "",
  "signUpStatus": "NOT_STARTED",
}
`;

exports[`auth reducer should handle sign in failure 1`] = `
Object {
  "resetPasswordErrorMessage": "",
  "resetPasswordStatus": "NOT_STARTED",
  "signInErrorMessage": "Unable to sign in, wrong credentials",
  "signInStatus": "ERROR",
  "signUpErrorMessage": "",
  "signUpStatus": "NOT_STARTED",
}
`;

exports[`auth reducer should handle sign in loading 1`] = `
Object {
  "resetPasswordErrorMessage": "",
  "resetPasswordStatus": "NOT_STARTED",
  "signInErrorMessage": "",
  "signInStatus": "LOADING",
  "signUpErrorMessage": "",
  "signUpStatus": "NOT_STARTED",
}
`;

exports[`auth reducer should handle sign in success 1`] = `
Object {
  "resetPasswordErrorMessage": "",
  "resetPasswordStatus": "NOT_STARTED",
  "signInErrorMessage": "",
  "signInStatus": "SUCCESS",
  "signUpErrorMessage": "",
  "signUpStatus": "NOT_STARTED",
}
`;

exports[`auth reducer should handle sign up failure 1`] = `
Object {
  "resetPasswordErrorMessage": "",
  "resetPasswordStatus": "NOT_STARTED",
  "signInErrorMessage": "",
  "signInStatus": "NOT_STARTED",
  "signUpErrorMessage": "Unable to sign up, invalid credentials",
  "signUpStatus": "ERROR",
}
`;

exports[`auth reducer should handle sign up loading 1`] = `
Object {
  "resetPasswordErrorMessage": "",
  "resetPasswordStatus": "NOT_STARTED",
  "signInErrorMessage": "",
  "signInStatus": "NOT_STARTED",
  "signUpErrorMessage": "",
  "signUpStatus": "LOADING",
}
`;

exports[`auth reducer should handle sign up success 1`] = `
Object {
  "resetPasswordErrorMessage": "",
  "resetPasswordStatus": "NOT_STARTED",
  "signInErrorMessage": "",
  "signInStatus": "NOT_STARTED",
  "signUpErrorMessage": "",
  "signUpStatus": "SUCCESS",
}
`;

exports[`auth reducer should return the initial state 1`] = `
Object {
  "resetPasswordErrorMessage": "",
  "resetPasswordStatus": "NOT_STARTED",
  "signInErrorMessage": "",
  "signInStatus": "NOT_STARTED",
  "signUpErrorMessage": "",
  "signUpStatus": "NOT_STARTED",
}
`;
